livekit-android-sdk/src/main/java/io/livekit/android/room/track/video/VideoSinkVisibility.kt

```diff
@@ -0,0 +1,136 @@
+package io.livekit.android.room.track.video
+
+import android.graphics.Rect
+import android.os.Handler
+import android.os.Looper
+import android.view.View
+import android.view.ViewTreeObserver
+import androidx.annotation.CallSuper
+import androidx.compose.ui.layout.LayoutCoordinates
+import io.livekit.android.room.track.Track
+import java.util.*
+
+abstract class VideoSinkVisibility : Observable() {
+    abstract fun isVisible(): Boolean
+    abstract fun size(): Track.Dimensions
+
+    /**
+     * This should be called whenever the visibility or size has changed.
+     */
+    fun notifyChanged() {
+        setChanged()
+        notifyObservers()
+    }
+
+    /**
+     * Called when this object is no longer needed and should clean up any unused resources.
+     */
+    @CallSuper
+    open fun close() {
+        deleteObservers()
+    }
+}
+
+class ComposeVisibility : VideoSinkVisibility() {
+    private var lastCoordinates: LayoutCoordinates? = null
+
+    override fun isVisible(): Boolean {
+        return (lastCoordinates?.isAttached == true && lastCoordinates?.size?.width != 0 && lastCoordinates?.size?.height != 0)
+    }
+
+    override fun size(): Track.Dimensions {
+        val width = lastCoordinates?.size?.width ?: 0
+        val height = lastCoordinates?.size?.height ?: 0
+        return Track.Dimensions(width, height)
+    }
+
+    fun onGloballyPositioned(layoutCoordinates: LayoutCoordinates) {
+        val lastVisible = isVisible()
+        val lastSize = size()
+        lastCoordinates = layoutCoordinates
+
+        if (lastVisible != isVisible() || lastSize != size()) {
+            notifyChanged()
+        }
+    }
+
+    fun onDispose() {
+        if (lastCoordinates == null) {
+            return
+        }
+        lastCoordinates = null
+        notifyChanged()
+    }
+}
+
+class ViewVisibility(private val view: View) : VideoSinkVisibility() {
+
+    private val handler = Handler(Looper.getMainLooper())
+    private val globalLayoutListener = object : ViewTreeObserver.OnGlobalLayoutListener {
+        val lastVisibility = false
+        val lastSize = Track.Dimensions(0, 0)
+
+        override fun onGlobalLayout() {
+            handler.removeCallbacksAndMessages(null)
+            handler.postDelayed({
```

Comment:
will this be debouncing the calls? or would it simply queue them up? we'd want a debounced behavior

also, when elements become visible, we should notify immediately to minimize delay of having no content showing. the delay should only be applied to when things are moving to invisible.